File: src/main/java/com/koddy/server/auth/domain/model/oauth/OAuthProvider.java
package com.koddy.server.auth.domain.model.oauth;

import com.koddy.server.auth.exception.AuthException;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.Arrays;

import static com.koddy.server.auth.exception.AuthExceptionCode.INVALID_OAUTH_PROVIDER;

@Getter
@RequiredArgsConstructor
public enum OAuthProvider {
    GOOGLE("google"),
    KAKAO("kakao"),
    ZOOM("zoom"),
    ;

    private final String value;

    public static OAuthProvider from(final String value) {
        return Arrays.stream(values())
                .filter(it -> it.value.equals(value))
                .findFirst()
                .orElseThrow(() -> new AuthException(INVALID_OAUTH_PROVIDER));
    }
}

File: src/main/java/com/koddy/server/auth/domain/model/oauth/OAuthProvider.kt
package com.koddy.server.auth.domain.model.oauth

import com.koddy.server.auth.exception.AuthException
import com.koddy.server.auth.exception.AuthExceptionCode.INVALID_OAUTH_PROVIDER

enum class OAuthProvider(
    val value: String,
) {
    GOOGLE("google"),
    KAKAO("kakao"),
    ZOOM("zoom"),
    ;

    companion object {
        @JvmStatic
        fun from(value: String): OAuthProvider {
            return entries.firstOrNull { it.value == value }
                ?: throw AuthException(INVALID_OAUTH_PROVIDER)
        }
    }
}
